### PR TITLE
Fix generated constants: f64::INFINITY & f64::NEG_ INFINITY

### PR DESCRIPTION
**Pull request by gabiganam**
https://github.com/rust-lang/rust-bindgen/issues/2853

### CODE REVIEW

File: bindgen/codegen/helpers.rs [hunk 1]
```diff
@@ -301,32 +301,27 @@ pub(crate) mod ast_ty {
         }
     }
 
-    pub(crate) fn float_expr(
-        ctx: &BindgenContext,
-        f: f64,
-    ) -> Result<TokenStream, ()> {
+    pub(crate) fn float_expr(f: f64) -> Result<TokenStream, ()> {
```
Comment (eifrah-aws) [inline bindgen/codegen/helpers.rs:304]: According to this https://rust-lang.github.io/rust-clippy/master/index.html#/legacy_numeric_constants
This fix should also be applied for `f32` et al

**Review by eifrah-aws — CHANGES_REQUESTED** [review]

**Review by madsmtm — COMMENTED** [review]
I'm not sure of Bindgen's MSRV policy for generated crates, but these constants require at least Rust 1.43.

**Comment by pvdrz** [comment]
> I'm not sure of Bindgen's MSRV policy for generated crates, but these constants require at least Rust 1.43.

If that's the case, there should be a new variant in `RustTarget` representing 1.43 and this change should be gated behind a feature that enables if the target version is larger or equal than 1.43

**Comment by gabiganam** [comment]
> > I'm not sure of Bindgen's MSRV policy for generated crates, but these constants require at least Rust 1.43.
> 
> If that's the case, there should be a new variant in `RustTarget` representing 1.43 and this change should be gated behind a feature that enables if the target version is larger or equal than 1.43

Updated the PR, hope this is what you meant.

**Comment by madsmtm** [comment]
> @gabiganam requested a review from @madsmtm

Sorry, I'm the wrong person to ask for this, I'm not a maintainer here.

**Review by pvdrz — CHANGES_REQUESTED** [review]
This looks good to me. Could you please add this change to the changelog?

**Review by pvdrz — APPROVED** [review]

**Pull request merged**